packages/kotti-ui/source/kotti-pagination/KtPagination.vue

```diff
@@ -1,163 +1,139 @@
+			<li :class="paginatorClasses(maximumPage)" @click="nextPage">
+				<i
+					class="yoco kt-pagination__page-button"
+					v-text="Yoco.Icon.CHEVRON_RIGHT"
+				/>
 			</li>
 		</ul>
 	</div>
 </template>
 
-<script>
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import { computed, defineComponent, ref } from '@vue/composition-api'
+
+import { makeProps } from '../make-props'
+
 import PaginationExpanded from './components/PaginationExpanded.vue'
 import PaginationFlexible from './components/PaginationFlexible.vue'
 import PaginationFractionated from './components/PaginationFractionated.vue'
+import { KottiPagination } from './types'
 
-export default {
+export default defineComponent<KottiPagination.PropsInternal>({
 	name: 'KtPagination',
 	components: {
 		PaginationExpanded,
 		PaginationFlexible,
 		PaginationFractionated,
 	},
-	props: {
-		adjacentAmount: { type: Number, default: 1 },
-		fixedWidth: { type: Boolean, default: false },
-		page: { type: Number, default: 1 },
-		pageSize: { type: Number, default: 10 },
-		pagingStyle: { type: String, default: 'expand' },
-		total: { type: Number, required: true },
+	props: makeProps(KottiPagination.propsSchema),
+	setup(props, { emit }) {
+		const currentPage = ref(props.page - 1)
+		const pageAmount = computed(() => Math.ceil(props.total / props.pageSize))
+		const maximumPage = computed(() => pageAmount.value - 1)
 
-		/**
-		 * @deprecated
-		 * Use :pagingStyle='fraction' instead
-		 */
-		fractionStyle: { type: Boolean, default: false },
-	},
-	data() {
 		return {
-			currentPage: this.page - 1,
+			paginationComponent: computed(() => {
+				const isFlexLogical = 2 * (props.adjacentAmount + 1) < pageAmount.value
+				switch (props.pagingStyle) {
+					case 'flex':
+						return !isFlexLogical || pageAmount.value < 2
+							? PaginationExpanded.name
+							: PaginationFlexible.name
+					case 'fraction':
+						return PaginationFractionated.name
+					default:
+						return PaginationExpanded.name
+				}
+			}),
+			currentPage,
+			maximumPage,
+			nextPage: () => {
+				if (currentPage.value === maximumPage.value) return
+				currentPage.value += 1
+				emit('nextPageClicked', currentPage.value + 1)
+			},
+			paginatorClasses: (page) => ({
+				'kt-pagination__page-item': true,
+				'kt-pagination__page-item--is-disabled': currentPage.value === page,
+			}),
+			paginationProps: computed(() => ({
+				adjacentAmount: props.adjacentAmount,
+				currentPage: currentPage.value,
+				fixedWidth: props.fixedWidth,
+				maximumPage: maximumPage.value,
+				pageSize: props.pageSize,
+				total: props.total,
+			})),
+			previousPage: () => {
+				if (currentPage.value === 0) return
+				currentPage.value -= 1
+				emit('previousPageClicked', currentPage.value + 1)
+			},
+			setCurrentPage: (page) => {
+				if (page === currentPage.value) return
+				currentPage.value = page
+				emit('currentPageChange', currentPage.value + 1)
```

Comment:
This API should probably be simplified. Why would the app care if the previous button was clicked instead of a specific page number? Let’s just emit the normal event

Maybe in a follow-up PR after this one is merged, alternatively let’s make an issue if you don’t want to do this immediately